pkg/vendir/cmd/sync.go

```diff
@@ -176,6 +177,9 @@ func (o *SyncOptions) Run() error {
 		return nil
 	}
 
+	if newLockConfig.IsEqualTo(existingLockConfig) {
```

Comment:
Couldn't this if statement be inside the `o.Locked` if statement?
Because if it is not locked the existingLockConfig will be empty struct which is not equal to the newLockConfig for sure

Reply:
Good question. I thought about where to put the check, because essentially, "if it's the same, don't write it" means we need to know three things:
1. Does the lock file even exist? Don't blow up checking an "old" lock file that doesn't exist
2. Did the lock file change? this means we need to read it sometime
3. Is the desired state of the lock file done updating?

Twice in the sync command we read the lock file. First in the `if o.Locked` block and once in the `if len(dirs) > 0` block near the end. I wanted to piggy-back on those if possible.

Maybe I'm being too clever with it and adding confusion. Let me try something simpler and see what you think.

Reply:
New change incoming...